### PR TITLE
Use user-installed tifffile if available

### LINKED ISSUE #3644: tifffile: try to use the one in the user's install first
Should we try importing tifffile before using the one we versionned it?

### PR DESCRIPTION
**Pull request by jni**
Fixes #3644

Note that we already have tifffile in our optional dependencies, so
these lines will be tested by one configuration or another of our test
matrix. (If I remember correctly, there's no easy way for coverage to be
measured across the whole matrix.)

- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [x] ~~[Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)~~
- [x] ~~Gallery example in `./doc/examples` (new features only)~~
- [x] ~~Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark~~
- [x] Unit tests

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/io/_plugins/tifffile_plugin.py [hunk 1]
```diff
@@ -1,4 +1,7 @@
-from ...external.tifffile import TiffFile, imsave
+try:
+    from tifffile import TiffFile, imsave
+except ImportError:
```
Comment (hmaarrfk) [inline skimage/io/_plugins/tifffile_plugin.py]: Does this work in 2.7 and 3.5?

**Comment by hmaarrfk** [comment]
Coverage has an option to combine reports from different cis

**Comment by sciunto** [comment]
then, shouldn't we test against the last version of tifffile in our CI?

**Comment by hmaarrfk** [comment]
tifffile 2019 is installed 
https://travis-ci.org/scikit-image/scikit-image/jobs/479729277#L2617

**Comment by hmaarrfk** [comment]
This may cause a conflict with https://github.com/scikit-image/scikit-image/pull/3675
so i'm going to wait to hear back on 3675 before merging this.

**Comment by jni** [comment]
:+1:

**Comment by hmaarrfk** [comment]
I think the question now is how to accomodate the different parameters in the various versions of tifffile.

cc: @cgohlke

**Comment by cgohlke** [comment]
The next version of tifffile will ignore the `fastij` and `multifile_close` parameters, and map the `pages` parameter to `key` in `tifffile.imread`. That should be (mostly) backwards compatible for skimage if the skimage tifffile plugin calls `tifffile.imread` instead of `with TiffFile... tif.save...`. Using both `pages` and `key` in `imread` will raise a `TypeError` as does passing `pages` to `TiffFile.__init__`.

**Comment by hmaarrfk** [comment]
Is the issue:

```
    if 'img_num' in kwargs:
        kwargs['key'] = kwargs.pop('img_num')
    # parse_kwargs will extract keyword arguments intended for the TiffFile 
    # class and remove them from the kwargs dictionary in-place
    tiff_keys = ['multifile', 'multifile_close', 'pages', 'fastij', 'is_ome']
    kwargs_tiff = parse_kwargs(kwargs, *tiff_keys)
```

I feel like the solution is to:
1. Add a deprecation warning to discourage the use of `img_num`?

scikit-image should not encourage the use of old APIs without a deprecation policy

**Comment by hmaarrfk** [comment]
I restarted the builds on travis. If they pass, I'll merge.

We can address compatibility with future tifffile in the future.

**Pull request merged**